testutil/network/basenet/accounts.go

```diff
@@ -0,0 +1,200 @@
+func (memnet *BaseInMemoryNetwork) FundOnChainAccounts(t *testing.T) {
+	t.Helper()
+
+	// NB: while it may initially seem like the memnet#Fund<actor>Accounts() methods
+	// can be refactored into a generic function, this is not possible so long as the genesis
+	// state lists are passed directly & remain a slice of concrete types (as opposed to pointers).
+	// Under these conditions, a generic function would not be able to unmarshal the genesis state
+	// stored in the in-memory network because it is unmarshalling uses reflection, and it is not
+	// possible to reflect over a nil generic type value.
+
+	// Retrieve the supplier module's genesis state from cosmos-sdk in-memory network.
+	supplierGenesisState := network.GetGenesisState[*suppliertypes.GenesisState](t, suppliertypes.ModuleName, memnet)
+	if supplierGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "supplier")
+	} else {
+		// Initialize on-chain accounts for genesis suppliers.
+		memnet.FundSupplierAccounts(t, supplierGenesisState.SupplierList...)
+
+	}
+
+	// Retrieve the application module's genesis state from cosmos-sdk in-memory network.
+	appGenesisState := network.GetGenesisState[*apptypes.GenesisState](t, apptypes.ModuleName, memnet)
+	if appGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "application")
+	} else {
+		// Initialize on-chain accounts for genesis applications.
+		memnet.FundAppAccounts(t, appGenesisState.ApplicationList...)
+	}
+
+	// Retrieve the gateway module's genesis state from cosmos-sdk in-memory network.
+	gatewayGenesisState := network.GetGenesisState[*gatewaytypes.GenesisState](t, gatewaytypes.ModuleName, memnet)
+	if gatewayGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "gateway")
+	} else {
+		// Initialize on-chain accounts for genesis gateways.
+		memnet.FundGatewayAccounts(t, gatewayGenesisState.GatewayList...)
+	}
+
+	// need to wait for the account to be initialized in the next block
+	require.NoError(t, memnet.GetNetwork(t).WaitForNextBlock())
+}
+
+// FundAddress initializes an Account address and sequence number with the auth module
+// by sending some tokens from the in-memory network  validator, to the address provided.
+// This is a necessary prerequesite in order for the account with the given address
+// to be able to submit valid transactions (i.e. pay tx fees).
+// NOTE: It DOES NOT associate a public key with the account. This will happen when a tx
+// which is signed by the account is broadcast to the network for the first time.
+func (memnet *BaseInMemoryNetwork) FundAddress(
+	t *testing.T,
+	addr types.AccAddress,
+) {
+	t.Helper()
+
+	signerAccountNumber := 0
+	clientCtx := memnet.GetClientCtx(t)
+	net := memnet.GetNetwork(t)
+	val := net.Validators[0]
+
+	args := []string{
+		fmt.Sprintf("--%s=true", flags.FlagOffline),
+		fmt.Sprintf("--%s=%d", flags.FlagAccountNumber, signerAccountNumber),
+		fmt.Sprintf("--%s=%d", flags.FlagSequence, memnet.NextValidatorTxSequenceNumber(t)),
```

Comment:
```suggestion
		// In-memory network account flags.
		fmt.Sprintf("--%s=true", flags.FlagOffline),
		fmt.Sprintf("--%s=%d", flags.FlagAccountNumber, signerAccountNumber),
		fmt.Sprintf("--%s=%d", flags.FlagSequence, memnet.NextValidatorTxSequenceNumber(t)),
```
Or something like that https://github.com/pokt-network/poktroll/pull/290/files#r1450631362